### PR TITLE
Add unit test for checkedListBox

### PR DESCRIPTION
**Pull request by Olina-Zhang**
Related https://github.com/dotnet/winforms/issues/10453

## Proposed changes

- Add unit tests for CheckedListBox to test its properties: DataSource, DrawMode, ItemHeight, ThreeDCheckBoxes, UseCompatibleTextRendering

- Add unit tests for CheckedListBox to test its events: DataSourceChanged, DisplayMemberChanged, ItemCheck, Click, MouseClick,  MeasureItem, ValueMemberChanged

## Customer Impact

- None

## Regression? 

- No

## Risk

- Minimal

## Test methodology 

- Unit tests

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/winforms/pull/11265)

### CODE REVIEW

File: src/System.Windows.Forms/tests/UnitTests/CheckedListBoxTests.cs [hunk 1]
```diff
@@ -114,6 +130,36 @@ public void CheckedListBox_ThreeDCheckBoxes(bool expected)
         Assert.Equal(expected, box.ThreeDCheckBoxes);
     }
 
+    [WinFormsTheory]
+    [BoolData]
+    public void CheckedListBox_ThreeDCheckBoxes_SetWithHandle_ReturnsExpected(bool expected)
+    {
+        using CheckedListBox box = new()
+        {
```
Comment (LeafShi1) [inline src/System.Windows.Forms/tests/UnitTests/CheckedListBoxTests.cs:138]: if this test goal is test the property with handle created, then no need to add this line here

File: src/System.Windows.Forms/tests/UnitTests/CheckedListBoxTests.cs [hunk 2]
```diff
@@ -114,6 +130,36 @@ public void CheckedListBox_ThreeDCheckBoxes(bool expected)
...
+            ThreeDCheckBoxes = expected
+        };
+
+        box.Items.Add("item1");
+        box.ThreeDCheckBoxes = expected;
+
+        box.ThreeDCheckBoxes.Should().Be(expected);
```
Comment (LeafShi1) [inline src/System.Windows.Forms/tests/UnitTests/CheckedListBoxTests.cs:160]: The validation looks unrelated to `box.Items.Add("item1");`

File: src/System.Windows.Forms/tests/UnitTests/CheckedListBoxTests.cs [hunk 3]
```diff
@@ -594,10 +640,276 @@ public void CheckedListBox_Remove_CheckedItems(bool createControl)
...
+
+        checkedListBox.ItemHeight = 50;
+        checkedListBox.ItemHeight.Should().Be(originalHeight);
+
+        checkedListBox.ItemHeight = -1;
+        checkedListBox.ItemHeight.Should().Be(originalHeight);
+    }
+
+    [WinFormsTheory]
+    [InlineData(true)]
```
Comment (SimonZhao888) [inline src/System.Windows.Forms/tests/UnitTests/CheckedListBoxTests.cs:708]: [BoolData]

File: src/System.Windows.Forms/tests/UnitTests/CheckedListBoxTests.cs [hunk 4]
```diff
@@ -594,10 +640,276 @@ public void CheckedListBox_Remove_CheckedItems(bool createControl)
...
+
+        checkedListBox.UseCompatibleTextRendering = value;
+        checkedListBox.UseCompatibleTextRendering.Should().Be(value);
+        checkedListBox.UseCompatibleTextRenderingInternal.Should().Be(value);
+    }
+
+    [WinFormsFact]
+    public void CheckedListBox_UseCompatibleTextRendering_SetWithSameValue_DoesNotChangeValue()
+    {
+        using CheckedListBox checkedListBox = new();
```
Comment (SimonZhao888) [inline src/System.Windows.Forms/tests/UnitTests/CheckedListBoxTests.cs:722]: This test case doesn't seem to make any sense.
We need to validate the default value of the control's public property.
Please add UseCompatibleTextRendering, UseCompatibleTextRendering and so on properties default checks to CheckedListBox_Constructor.

File: src/System.Windows.Forms/tests/UnitTests/CheckedListBoxTests.cs [hunk 5]
```diff
@@ -594,10 +640,276 @@ public void CheckedListBox_Remove_CheckedItems(bool createControl)
...
+        {
+            callCount++;
+            sender.Should().Be(checkedListBox);
+            e.Should().Be(EventArgs.Empty);
+        };
+
+        // Add and remove.
+        checkedListBox.DataSourceChanged += handler;
+        checkedListBox.DataSourceChanged -= handler;
+        checkedListBox.OnDataSourceChanged(EventArgs.Empty);
```
Comment (SimonZhao888) [inline src/System.Windows.Forms/tests/UnitTests/CheckedListBoxTests.cs:744]: We also need add the following checks:

checkedListBox.DataSourceChanged += handler;
checkedListBox.OnDataSourceChanged(EventArgs.Empty);
callCount.Should().Be(1);

checkedListBox.DataSourceChanged -= handler;
checkedListBox.OnDataSourceChanged(EventArgs.Empty);
callCount.Should().Be(1);
       
And make this change to the following cases.

File: src/System.Windows.Forms/tests/UnitTests/CheckedListBoxTests.cs (hunk 2 above)
Comment (Olina-Zhang) [reply src/System.Windows.Forms/tests/UnitTests/CheckedListBoxTests.cs:145]: scenario is for checkedlistBox with item added

File: src/System.Windows.Forms/tests/UnitTests/CheckedListBoxTests.cs (hunk 4 above)
Comment (Olina-Zhang) [reply src/System.Windows.Forms/tests/UnitTests/CheckedListBoxTests.cs]: Moved the scenario of checking default UseCompatibleTextRendering value in CheckedListBox_Constructor. The expected value should be `false` because when adding a checkedListBox to form designer, that property is `false`, but in unit test, the actual value is `true`.

File: src/System.Windows.Forms/tests/UnitTests/CheckedListBoxTests.cs [hunk 6]
```diff
@@ -594,10 +625,242 @@ public void CheckedListBox_Remove_CheckedItems(bool createControl)
         }
     }
 
+    [WinFormsFact]
+    public void CheckedListBox_DataSource_GetSet_ReturnsExpected()
+    {
+        using CheckedListBox checkedListBox = new();
+        var dataSource1 = new List<string> { "item1", "item2", "item3" };
+        var dataSource2 = new List<string> { "item4", "item5", "item6" };
```
Comment (lonitra) [inline src/System.Windows.Forms/tests/UnitTests/CheckedListBoxTests.cs:633]:
```suggestion
        List<string> dataSource1 = ["item1", "item2", "item3" ];
        List<string> dataSource2 = ["item4", "item5", "item6" ];
```

File: src/System.Windows.Forms/tests/UnitTests/CheckedListBoxTests.cs [hunk 7]
```diff
@@ -594,10 +625,242 @@ public void CheckedListBox_Remove_CheckedItems(bool createControl)
...
+        using SubCheckedListBox checkedListBox = new();
+        int callCount = 0;
+        ItemCheckEventHandler handler = (sender, e) =>
+        {
+            callCount++;
+            sender.Should().Be(checkedListBox);
+            e.Should().NotBeNull();
+        };
+
+        ItemCheckEventArgs eventArgs = new ItemCheckEventArgs(0, CheckState.Checked, CheckState.Unchecked);
```
Comment (lonitra) [inline src/System.Windows.Forms/tests/UnitTests/CheckedListBoxTests.cs:746]: Please use target type new
```suggestion
        ItemCheckEventArgs eventArgs = new(0, CheckState.Checked, CheckState.Unchecked);
```

File: src/System.Windows.Forms/tests/UnitTests/CheckedListBoxTests.cs [hunk 8]
```diff
@@ -594,10 +625,242 @@ public void CheckedListBox_Remove_CheckedItems(bool createControl)
...
+        using SubCheckedListBox checkedListBox = new();
+        int callCount = 0;
+        MouseEventHandler handler = (sender, e) =>
+        {
+            callCount++;
+            sender.Should().Be(checkedListBox);
+            e.Should().NotBeNull();
+        };
+
+        MouseEventArgs eventArgs = new MouseEventArgs(MouseButtons.Left, 1, 0, 0, 0);
```
Comment (lonitra) [inline src/System.Windows.Forms/tests/UnitTests/CheckedListBoxTests.cs:790]:
```suggestion
        MouseEventArgs eventArgs = new(MouseButtons.Left, 1, 0, 0, 0);
```

File: src/System.Windows.Forms/tests/UnitTests/CheckedListBoxTests.cs [hunk 9]
```diff
@@ -594,10 +625,242 @@ public void CheckedListBox_Remove_CheckedItems(bool createControl)
...
+        int callCount = 0;
+        MeasureItemEventHandler handler = (sender, e) =>
+        {
+            callCount++;
+            sender.Should().Be(checkedListBox);
+            e.Should().NotBeNull();
+        };
+
+        using Bitmap image = new(10, 10);
+        MeasureItemEventArgs eventArgs = new MeasureItemEventArgs(Graphics.FromImage(image), 1);
```
Comment (lonitra) [inline src/System.Windows.Forms/tests/UnitTests/CheckedListBoxTests.cs:814]:
```suggestion
        MeasureItemEventArgs eventArgs = new(Graphics.FromImage(image), 1);
```

**Review by lonitra — COMMENTED** [review]
Some small comments, LGTM otherwise 👍

**Review by lonitra — APPROVED** [review]
